src/main/scala/scalaz/stream/Process.scala

```diff
@@ -871,6 +871,24 @@ object Process extends ProcessInstances {
   def iterateEval[F[_], A](start: A)(f: A => F[A]): Process[F, A] =
     emit(start) ++ await(f(start))(iterateEval(_)(f))
 
+  /**
+   * Create an iterator to use as a source for a `Process`,
+   * which lazily emits the values of the iterator, then halts.
+   *
+   * Be sure that iteratorCreator creates any mutable or external
+   * resources that the iterator uses are created as part of the
+   * Task. Also, do not use Task.now to create the iterator.
+   */
+  def iterator[F[_], O](iteratorCreator: F[Iterator[O]]): Process[F, O] = {
+    //This design was based on unfold.
+    def go(iterator: Iterator[O]): Process0[O] = {
```

Comment:
Maybe this is just me being a purist, but given that both `hasNext` and `next` are effectful, it seems to me that the following would be a preferable implementation:

```scala
await(iteratorCreator) { iterator =>
  val hasNext = Task delay { iterator.hasNext }
  val next = Task delay { iterator.next() }

  await(hasNext) { hn => if (hn) eval(next) else halt } repeat
}
```

The downside here is that we're restricting the effect to be `Task`, but I'm not sure that's necessarily a bad thing.  If someone instantiates `F` to be an eager effect, things are going to be highly screwy.  Also, and perhaps more importantly, evaluation of processes which do *not* contain `Await` nodes is *eager* in `stepAsync`, and so the iterator is going to get pulled much faster than you might expect.